### PR TITLE
update goreleaser template

### LINKED ISSUE #70: Mismatches in the files in `deployment-templates`
### What happened?

I recently created a custom provider based on this template and i ran into a few issues.

## 1. Make command mismatch in `deployment-templates/.goreleaser.yml`

It is `build_provider`, but it should be `provider`.

https://github.com/pulumi/pulumi-tf-provider-boilerplate/blob/efb5bdb0aa7a6f5fad9555c98bd43fa700c21eee/deployment-templates/.goreleaser.yml#L6

https://github.com/pulumi/pulumi-tf-provider-boilerplate/blob/efb5bdb0aa7a6f5fad9555c98bd43fa700c21eee/Makefile#L55

## 2. Binary name mismatch in `deployment-templates/.goreleaser.yml`

It is `provider-xyz` but it should be `pulumi-resource-xyz`, the latter is the file name the pulumi plugin install command looks for.

https://github.com/pulumi/pulumi-tf-provider-boilerplate/blob/efb5bdb0aa7a6f5fad9555c98bd43fa700c21eee/deployment-templates/.goreleaser.yml#L8

https://github.com/pulumi/pulumi-tf-provider-boilerplate/blob/efb5bdb0aa7a6f5fad9555c98bd43fa700c21eee/deployment-templates/.goreleaser.yml#L22

## 3. Job name mismatch in `deployment-templates/release.yml`

There is a job named `publish-binary` but the other jobs have `needs: publish_binary` 

https://github.com/pulumi/pulumi-tf-provider-boilerplate/blob/efb5bdb0aa7a6f5fad9555c98bd43fa700c21eee/deployment-templates/release.yml#L32

https://github.com/pulumi/pulumi-tf-provider-boilerplate/blob/efb5bdb0aa7a6f5fad9555c98bd43fa700c21eee/deployment-templates/release.yml#L63

### Steps to reproduce

Copy the files in `deployment-templates` according to the readme and try to publish a release.

### Expected Behavior

A release should be published.

### Actual Behavior

Github actions will crash.

### Versions used

_No response_

### Additional context

_No response_

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

### PR DESCRIPTION
**Pull request by jkisk**
Fixes #70, part of which was already resolved by #71

### CODE REVIEW

File: deployment-templates/.goreleaser.yml [hunk 1]
```diff
@@ -3,9 +3,9 @@ archives:
     name_template: '{{ .Binary }}-{{ .Tag }}-{{ .Os }}-{{ .Arch }}'
 before:
   hooks:
-    - make build_provider
+    - make provider
```
Comment (stack72) [inline deployment-templates/.goreleaser.yml:6]: we only need to run make tfgen before the provider

Comment (jkisk) [reply deployment-templates/.goreleaser.yml:6]: oh, I'll update that, thank you

Comment (stack72) [reply deployment-templates/.goreleaser.yml:6]: make tfgen creates the schema which is required for goreleaser to make it's build of the provider

Running make provider again would just duplicate the work :)

File: deployment-templates/.goreleaser.yml [hunk 2]
```diff
@@ -3,9 +3,9 @@ archives:
     name_template: '{{ .Binary }}-{{ .Tag }}-{{ .Os }}-{{ .Arch }}'
 before:
   hooks:
-    - make build_provider
+    - make tfgen
 builds:
-  - binary: provider-xyz
+  - binary: provider-resource-xyz
```
Comment (ringods) [inline deployment-templates/.goreleaser.yml:8]: @jkisk just bumped into an issue with @rawkode. The binary name needs to be:

`pulumi-resource-xyz`

not 

`provider-resource-xyz`

Can you fix this?

Comment (stack72) [reply deployment-templates/.goreleaser.yml:8]: Feel free to send a PR @ringods - will happily merge it :)

Comment (ringods) [reply deployment-templates/.goreleaser.yml:8]: @stack72 https://github.com/pulumi/pulumi-tf-provider-boilerplate/pull/83

**Pull request merged**